### DISCUSSION TITLE
Deprecated form field userAgent, and extraHttpHeaders

### DISCUSSION THREAD

**Discussion by LFaugeron** [gotenberg/gotenberg#842] [Q&A]
Hello,
I'm trying to bump the version of my gotenberg service to v8. I'm having an issue.

I was using the userAgent form field to change the userAgent and detect if the request was coming from my gotenberg or not on my website.

Since the field is deprecated, is there any replacement ? I supposed, I could do it by adding extraHttpHeaders like this in python:
`"extraHttpHeaders": f'{{"User-Agent":"my-agent"}}'`

But it is not working. If anyone has an idea, I would love some help please.

Thanks

**Comment by gulien** [comment]
Another option would be to set a custom header via `extraHttpHeaders` and use this custom header instead 🤔

**Reply by LFaugeron** [reply]
It won't work in my case, but is it possible to send cookies in extraHttpHeaders like this ? 
`"extraHttpHeaders": f'{{"cookie":"my-agent"}}'`

**Reply by gulien** [reply]
No, it’s a feature in my TODO to be able to send cookies 😅

**Comment by wjkoh** [comment]
If I understand correctly, there is currently no way to set custom `User-Agent`, `Accept-Encoding`, and `Host` headers. Is that right? Some U.S. government websites that allow crawling require crawlers to set their User-Agent to "CompanyName [company@example.com](mailto:company@example.com)" to properly rate-limit each crawler. Is there any workaround to this problem?

**Reply by gulien** [reply]
@wjkoh `extraHttpHeaders`?

**Reply by wjkoh** [reply]
I tried to generate a PDF using a Gotenberg container, but the resulting file was completely blank. I sent a POST request with [this URL](https://www.sec.gov/Archives/edgar/data/1067983/000119312524133049/d759664d8k.htm), but all I got was a white, empty page. You need to set the HTTP headers like [this](https://www.sec.gov/os/accessing-edgar-data). However, when I manually downloaded the HTML file and sent it to my Gotenberg container, the conversion worked like a charm. This made me think that maybe the `extraHttpHeaders` parameter isn't working as it should for `User-Agent`, `Accept-Encoding` and `Host`.

To get to the bottom of this, I was wondering if there's a way to take a closer look at the actual headers and body that a Gotenberg instance sends when it makes a GET request.

**Reply by galenwarren** [reply]
It looks ```Accept``` can't be set via ```extraHttpHeaders``` either?

**Comment by gulien** [comment]
@LFaugeron see #901
